### ISSUE TITLE
PS 1.7 PASSWORD_LENGTH

### ISSUE THREAD

**Issue by anael687** [PrestaShop/PrestaShop#10475]
**Describe the bug**
Hi, I've tried to change PASSWORD_LENGTH into classes/Validate.php (to 8 for example). But when registering a new customer with Classic theme, the allowed length is still 5. I had a look into the html code, on the register page, and saw a pattern=".{5,}". If I change the value 5 by 1, I'm able to register a new customer with a password of 1 character. So it seems that function isPasswd(...) is never used on the controller, because there is no check on the backend of the length of the password.

**To Reproduce**
1. Try changing password_length to 8 in classes/Validate.php
2. Try registering a new customer with a 5 char password
3. it will work

**Screenshots**
If applicable, add screenshots or screenrecords to help explain your problem.

**Additionnal information**
PrestaShop version: 1.7.4.2
PHP version: 7.0.31

**Comment by khouloudbelguith** [comment]
Hi @anael687,

In the Project_folder/themes/Classic/templates/_partials/form-fiels.tpl file, you need to replace this 
```
{block name='form_field_item_password'}
          <div class="input-group js-parent-focus">
            <input
              class="form-control js-child-focus js-visible-password"
              name="{$field.name}"
              type="password"
              value=""
              pattern=".{literal}{{/literal}5,{literal}}{/literal}"
              {if $field.required}required{/if}
            >
```

By this
```
{block name='form_field_item_password'}
          <div class="input-group js-parent-focus">
            <input
              class="form-control js-child-focus js-visible-password"
              name="{$field.name}"
              type="password"
              value=""
              pattern=".{literal}{{/literal}8,{literal}}{/literal}"
              {if $field.required}required{/if}
            >
```

I close the issue, feel free topen a new one when needed.
Ps: you need to clear cache from your browser & your shop.

Best regards, Khouloud

**Comment by anael687** [comment]
Thank for the prompt answer. But what about the fact it is not checked with Validate.php ?

**Comment by khouloudbelguith** [comment]
Hi @anael687,

It is to check the contact form about creating a new customer in BO.

Best regards, Khouloud

**Comment by web-fu** [comment]
I've found the same problem, and I fixed it modifying classes/form/CustomerForm.php adding at line 130:
```
        $passwordField = $this->getField('password');
        if (!Validate::isPasswd($passwordField->getValue())) {
            $passwordField->addError($this->translator->trans(
                'The password is not in a valid format.', array(), 'Shop.Notifications.Error'
            ));
        }
```

I prefer server security against Front End security

**Comment by anael687** [comment]
Thank you for sharing. Much appreciated.

**Comment by matks** [comment]
@khouloudbelguith I think this is an issue we should consider, we indeed cannot rely on front-end validation that can be easily bypassed by using a cURL client.

@web-fu do you feel like making a PR with your contribution ? I can do it, but I would "steal" the credit you deserve 😉

**Comment by web-fu** [comment]
Submitted!
I almost forgot this fix 👍

**Comment by colinegin** [comment]
Hello @PierreRambaud , is this issue still valid since a PR improving the password policy has already been merged ?

**Comment by PierreRambaud** [comment]
The PR about the password policy has not been merged https://github.com/PrestaShop/PrestaShop/pull/12824 still waiting for a decision to know what we want.
Still thinking it should be merged.

**Comment by florine2623** [comment]
@matks ,

The function mentioned by @anael687 is deprecated.
<img width="682" alt="Screenshot 2023-10-18 at 17 02 49" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/3068b98d-bb16-4145-a496-25598b0e0a10">

What is the future of the issue ?

**Comment by matks** [comment]
In PrestaShop 8.0 a new password policy was introduced and it allows shop admin to choose how secure front-end passwords should be. This fixes this issue for me 👍 consider it fixed since 8.0.0

Details about the password policy https://www.prestashop-project.org/releases/prestashop80/
